helix-term/src/commands/typed.rs

```diff
@@ -1606,6 +1606,100 @@ fn run_shell_command(
     Ok(())
 }
 
+fn read_file_info_buffer(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    let (view, doc) = current!(cx.editor);
+
+    ensure!(!args.is_empty(), "file name is expected");
+
+    let filename = args.get(0).unwrap();
+    let path = PathBuf::from(filename.to_string());
+    if path.exists() {
+        match std::fs::File::open(path) {
+            Ok(file) => {
+                let mut reader = BufReader::new(file);
+
+                match from_reader(&mut reader, doc.encoding()) {
+                    Ok(contents) => {
+                        let contents = Tendril::from(contents);
+                        let selection = doc.selection(view.id);
+                        let transaction = Transaction::insert(doc.text(), selection, contents);
+                        doc.apply(&transaction, view.id);
+                    }
+                    Err(error) => cx
+                        .editor
+                        .set_error(format!("error reading file: {}", error)),
+                }
+            }
+            Err(error) => cx
+                .editor
+                .set_error(format!("error reading file: {}", error)),
+        }
+    } else {
+        cx.editor
+            .set_error(format!("file doesn't exist: {}", filename));
+    }
```

Comment:
The statusline will show an error if you return the `Err` variant of an `anyhow::Result`, so you can remove all of these levels of nesting by returning early:

```rs
if !path.exists() {
    bail!("file doesn't exist: {}", filename);
}

let file = std::fs::File::open(path).map_err(|err| anyhow!("error reading file: {}", err))?;

// ...
```

Reply:
Thanks! For some reason, I assumed these methods need to return `Ok(())` even if they errored unless it's not a catastrophic error. I ended up with a lot of nesting. I have cleaned it all now.